### PR TITLE
docs: Data model for append-only Merkle trees

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
This change adds some terminology to the docs.

### CODE REVIEW

**Comment by pavelkalinnikov** [comment]
https://github.com/pavelkalinnikov/merkle/blob/document_merkle_data_model/docs/data_model.md

**Review by hickford — APPROVED** [review]

File: docs/data_model.md [hunk 1]
```diff
@@ -1,13 +1,46 @@
...
+nodes that recursively depend on these entries, are updated. While the nodes can
+be updated, they are in **ephemeral** state. Eventually, when the log grows past
+a certain size, a node becomes **perfect**, and is never modified again.
+
+Effectively, perfect nodes are immutable / write-once registers, and ephemeral
+nodes are mutable.
+
+### Tree state
+
+To represent the state of the entire log, often a single ephemeral node is used
```
Comment (AlCutter) [inline docs/data_model.md:38]: How about rearranging slightly to something like:

> To represent the state of the entire log the single node which covers all leaves in the log is used. For example, in a tree with 21 ... node `5.0`.

File: docs/data_model.md [hunk 2]
```diff
@@ -1,13 +1,46 @@
...
+which covers all the corresponding leaves. For example, in a tree with 21
+leaves, as in the picuture above, this would be the ephemeral node 5.0.
```
Comment (AlCutter) [inline docs/data_model.md:40]: typo: picture

**Review by AlCutter — COMMENTED** [review]
Thanks dude, few minor bits inline.

**Review by AlCutter — APPROVED** [review]

**Comment by pavelkalinnikov** [comment]
I did not address all rewording comments, was going to do it later. But anyway, this doc needs further work, so can be addressed in follow-ups.

**Comment by AlCutter** [comment]
> I did not address all rewording comments, was going to do it later. But anyway, this doc needs further work, so can be addressed in follow-ups.

I noticed, but interpreted it as declining the suggestions - follow up(s) are, of course, more than welcome, though :)

**Pull request merged**